ui/js/page/settings/view.jsx

```diff
@@ -85,6 +85,12 @@ class SettingsPage extends React.PureComponent {
     this.props.setClientSetting("showNsfw", event.target.checked);
   }
 
+  onLanguageChange(language) {
+    lbry.setClientSetting('language', language);
+    window.app.i18n.setLocale(language);
+    this.setState({language: language})
```

Comment:
After the refactor, language would be a property, not a part of the state. And the language setting would happen inside of an action. In this model, components never change the state of the app directly, they only dispatch actions.